transformer_engine/common/fused_attn/fused_attn_f16_arbitrary_seqlen.cu

```diff
@@ -587,17 +739,30 @@ void fused_attn_arbitrary_seqlen_fwd_impl(
             auto sAfterMaskTensor = createCausalMask(
                                 b, h, s_q, s_kv, d, layout, tensorType, &ops, sScaleTensor);
 
+            std::shared_ptr<cudnn_frontend::Tensor> softmaxInput;
+
+            if (variable_sequence_length) {
+                auto sAfterPaddingMaskTensor = createPaddingMask(
+                    b, h, s_q, s_kv, d, layout, tensorType, &ops, sAfterMaskTensor);
+                softmaxInput = std::make_shared<cudnn_frontend::Tensor>(
+                    std::move(sAfterPaddingMaskTensor));
+            } else {
+                softmaxInput = std::make_shared<cudnn_frontend::Tensor>(
+                    std::move(sAfterMaskTensor));
+            }
```

Comment:
Nitpick: `unique_ptr` is somewhat preferred if we're not sharing:
```suggestion
            std::unique_ptr<cudnn_frontend::Tensor> softmaxInput;

            if (variable_sequence_length) {
                auto sAfterPaddingMaskTensor = createPaddingMask(
                    b, h, s_q, s_kv, d, layout, tensorType, &ops, sAfterMaskTensor);
                softmaxInput = std::make_unique<cudnn_frontend::Tensor>(
                    std::move(sAfterPaddingMaskTensor));
            } else {
                softmaxInput = std::make_unique<cudnn_frontend::Tensor>(
                    std::move(sAfterMaskTensor));
            }
```

Reply:
Good suggestion. Thanks